src/hato/multipart.clj

```diff
@@ -55,12 +84,94 @@
        (take 30)
        (apply str "hatoBoundary")))
 
-(defn body
-  "Returns an InputStream from the multipart inputs.
+(defprotocol MultipartParam
```

Comment:
The following protocol defines the possible multiple param content types (can be easily extended). We need to be able to coerce the param into an input stream and determine it's length (if possible). For types that have potentially unknown lengths (such as an InputStream), a client can set it by supplying :content-length in the multipart param map.